diff --git a/node_modules/react-native/scripts/cocoapods/spm.rb b/node_modules/react-native/scripts/cocoapods/spm.rb
index 0b13207..650c153 100644
--- a/node_modules/react-native/scripts/cocoapods/spm.rb
+++ b/node_modules/react-native/scripts/cocoapods/spm.rb
@@ -31,6 +31,10 @@ class SPMManager
         )
         log " Adding workaround for Swift package not found issue"
         target = project.targets.find { |t| t.name == pod_name}
+        if target.nil?
+          log_warning "Target #{pod_name} not found in Pods project. Skipping SPM workaround."
+          next
+        end
         target.build_configurations.each do |config|
           target.build_settings(config.name)['SWIFT_INCLUDE_PATHS'] ||= ['$(inherited)']
           search_path = '${SYMROOT}/${CONFIGURATION}${EFFECTIVE_PLATFORM_NAME}/'
@@ -66,6 +70,8 @@ class SPMManager
   end
 
   def add_spm_to_target(project, target, url, requirement, products)
+    return if target.nil?
+
     pkg_class = Xcodeproj::Project::Object::XCRemoteSwiftPackageReference
     ref_class = Xcodeproj::Project::Object::XCSwiftPackageProductDependency
     pkg = project.root_object.package_references.find { |p| p.class == pkg_class && p.repositoryURL == url }
@@ -76,6 +82,9 @@ class SPMManager
       log(" Adding package to workspace: #{pkg.inspect}")
       project.root_object.package_references << pkg
     end
+
+    target.package_product_dependencies ||= []
+
     products.each do |product_name|
       ref = target.package_product_dependencies.find do |r|
         r.class == ref_class && r.package == pkg && r.product_name == product_name
